nil/services/nilservice/service.go

```diff
@@ -453,11 +459,18 @@ func createNetworkManager(ctx context.Context, cfg *Config) (*network.Manager, e
 	return network.NewManager(ctx, cfg.Network)
 }
 
+type shards struct {
```

Comment:
IMO `shards` is too common name. Is it possible to choose something more specific?